### PR TITLE
lib-user: Add project card component

### PR DESCRIPTION
**Pull request by mcbouslog**
## Package
- lib-user

## Describe your changes
- add ProjectCard component
- add ProjectCard stories
- add ProjectCard tests
- move ProfileHeader component to shared folder
- refactor package.json for CI tests and to build and deploy lib-user storybook

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What user actions should my reviewer step through to review this PR?
  - review [Figma ProjectCard design here](https://www.figma.com/file/qbqbmR3t5XV6eKcpuRj7mG/Group-Stats?type=design&node-id=768-6524&mode=design&t=eak08Swc9mKME1QO-0), I will review Sean as well
- Which storybook stories should be reviewed?
  - http://localhost:6008/?path=/story/components-shared-projectcard--default (toggle views for small, medium, and large widths)
- Are there plans for follow up PR’s to further fix this bug or develop this feature? I think refactor with Next Link, add classification count stat in upper right

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## New Feature
- [ ] The PR creator has listed user actions to use when testing the new feature
- [ ] Unit tests are included for the new feature
- [ ] A storybook story has been created or updated
  - Example of SlideTutorial [component](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SlideTutorial/SlideTutorial.js) with docgen comments, and its [story](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/docs/other-slidetutorial--default)

### CODE REVIEW

**Comment by mcbouslog** [comment]
Noting per design review with Sean, I'm going to work on the following items:
- project image should be square
- add transition ease for hover/focus description show
- text content expand to fixed height
- add overflow scroll for longer project descriptions

**Comment by mcbouslog** [comment]
I'm not sure what to do about keyboard focus.
Currently, these changes do not include keyboard focus for the project description, which is relevant if the description overflows and a user would want to scroll to see the complete description.
Here's a video of the current changes in this PR:

https://github.com/zooniverse/front-end-monorepo/assets/12118751/cae8652d-162f-4419-ac95-3963b540765a

However, with some CSS and adding tabIndex to the description I can add keyboard focus, but then that element is always in the tab index order, and the content is announced twice with a screen reader (once when the card is focused, again when the text element is focused). Here's a video:

https://github.com/zooniverse/front-end-monorepo/assets/12118751/5ca401e4-3d28-4691-a6b5-41875d5d3cf1

I'm not sure which is preferable or if there's a different solution? The vast majority of project descriptions aren't that long, so this is related to an edge case.

File: packages/lib-user/src/components/shared/ProjectCard/ProjectCard.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        >
+          {displayName}
+        </SpacedText>
+        <StyledSpacedText
+          color={{ dark: 'neutral-6', light: 'dark-5' }}
+          size='inherit'
+          textAlign='center'
+          uppercase={false}
+        >
+          {description}
```
Comment (mcbouslog) [inline packages/lib-user/src/components/shared/ProjectCard/ProjectCard.js:102]: I'm not sure what to do about keyboard focus for the project description.
Currently, these changes do not include keyboard focus for the project description, which is relevant if the description overflows and a user would want to scroll to see the complete description.
Here's a video of the current changes in this PR:

https://github.com/zooniverse/front-end-monorepo/assets/12118751/cae8652d-162f-4419-ac95-3963b540765a

However, with some CSS and adding tabIndex to the description I can add keyboard focus, but then that element is always in the tab index order, and the content is announced twice with a screen reader (once when the card is focused, again when the text element is focused). Here's a video:

https://github.com/zooniverse/front-end-monorepo/assets/12118751/5ca401e4-3d28-4691-a6b5-41875d5d3cf1

I'm not sure which is preferable or if there's a different solution? The vast majority of project descriptions aren't that long, so this is related to an edge case.

Comment (mcbouslog) [reply packages/lib-user/src/components/shared/ProjectCard/ProjectCard.js:113]: Discussed on Slack and concluded to use line-clamp to truncate project name at 3 lines and project description at 4 lines.

File: packages/lib-user/src/components/shared/ProjectCard/ProjectCard.stories.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    imageSrc: 'https://panoptes-uploads.zooniverse.org/project_avatar/252e7e80-949b-4324-8af4-8d3ad7aafbde.jpeg'
+  }
+}
+
+export const WildwatchKenya = {
+  args: {
+    description: 'We need your help to count, identify, and track the giraffes and other wildlife living in our field conservation sites in northern Kenya! We need your help to count, identify, and track the giraffes and other wildlife living in our field conservation sites in northern Kenya!',
+    displayName: 'Deciphering Secrets: Unlocking the Manuscripts of Medieval Spain With Extra Text',
+    href: 'https://www.zooniverse.org/projects/sandiegozooglobal/wildwatch-kenya',
+    imageSrc: 'https://panoptes-uploads.zooniverse.org/project_avatar/5dc89775-a658-4d3a-8295-4d48903e142d.jpeg'
```
Comment (goplayoutside3) [inline packages/lib-user/src/components/shared/ProjectCard/ProjectCard.stories.js:65]: This story's name is Wildwatch Kenya, but the project card title is Deciphering Secrets - just a note in case you meant to use data from only one example project.

**Review by goplayoutside3 — APPROVED** [review]
Looks good! Just a couple of notes about CSS for future consideration.

File: packages/lib-user/src/components/shared/ProjectCard/ProjectCard.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    <StyledProjectCard
+      elevation='small'
+      forwardedAs='a'
+      href={href}
+      round='xxsmall'
+    >
+      <StyledProjectImage
+        background={{
+          image: `url(${imageSrc})`,
+          position: 'absolute',
```
[the remark this answers is not in the document]
Comment (mcbouslog) [reply packages/lib-user/src/components/shared/ProjectCard/ProjectCard.js]: Nice catch! I think `top` gets the desired outcome. Per Sean, the key for the image is that it is square. Relatedly, I adjusted the StyledProjectContent to 20%.

File: packages/lib-user/src/components/shared/ProjectCard/ProjectCard.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    height: 240px;
+    width: 189px;
+  }
+
+  @media (width <= 768px) {
+    font-size: 10px;
+    height: 200px;
+    width: 157px;
+  }
+`
```
[the remark this answers is not in the document]
Comment (mcbouslog) [reply packages/lib-user/src/components/shared/ProjectCard/ProjectCard.js:23]: Good question. I think the gap, wrap, padding, or similar styling can be left to the container/section (i.e. Top Projects)? Though certainly can revisit the ProjectCard component to be more dynamically styled or support a prop that allows for more dynamically styling. I'm not sure what specific change to make at this point in time, but can certainly revisit as the sections that will contain the ProjectCard are implemented and any issue surfaces.

File: packages/lib-user/src/components/shared/ProjectCard/ProjectCard.stories.js (hunk 2 above)
Comment (mcbouslog) [reply packages/lib-user/src/components/shared/ProjectCard/ProjectCard.stories.js]: Yes! I edited while demoing and testing edge cases, have edited again as Wildwatch Kenya then added a bunch of extra text to still test an edge case.
